ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/ConnectorInfoEntity.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ConnectorInfoEntity extends KsqlEntity {
+
+  private final ConnectorInfo info;
+  private final Action action;
+
+  @JsonCreator
+  public ConnectorInfoEntity(
+      @JsonProperty("statementText") final String statementText,
+      @JsonProperty("info") final ConnectorInfo info,
+      @JsonProperty("action") final Action action
+  ) {
+    super(statementText);
+    this.info = Objects.requireNonNull(info, "info");
+    this.action = Objects.requireNonNull(action, "action");
+  }
+
+  public ConnectorInfo getInfo() {
+    return info;
+  }
+
+  public Action getAction() {
+    return action;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final ConnectorInfoEntity that = (ConnectorInfoEntity) o;
+    return Objects.equals(info, that.info);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(info);
+  }
+
+  @Override
+  public String toString() {
+    return "ConnectorInfoEntity{"
+        + "info=" + info
+        + '}';
+  }
+
+  public enum Action {
```

Comment:
What's the point in the action enum?   Only `CREATE` seems to be used, and the CLI rejects anything that's not `CREATE`....?

If the plan is to output more information on a `DESCRIBE` call vs the `CREATE` call output just the name, then I would argue they should be two different response classes.

Reply:
okay - I was worried it would cause an explosion of response types but i'm not too picky